src/Auth_Command.php

```diff
@@ -0,0 +1,332 @@
+
+		if ( $user ) {
+			EE::exec( sprintf( 'docker exec %s htpasswd -D /etc/nginx/htpasswd/%s %s', EE_PROXY_TYPE, $this->db->site_url, $user ), true, true );
+		} else {
+			$this->fs->remove( EE_CONF_ROOT . '/nginx/htpasswd/' . $file );
+			EE::log( sprintf( 'http auth removed for `%s` scope', $this->db->site_url ) );
+		}
+		$this->reload();
+	}
+
+	/**
+	 * Lists http auth users of a site.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [<site-name>]
+	 * : Name of website.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - yaml
+	 *   - json
+	 *   - count
+	 *   - text
+	 * ---
+	 */
+	public function list( $args, $assoc_args ) {
+
+		$global = $this->populate_info( $args, __FUNCTION__ );
+		$file   = EE_CONF_ROOT . '/nginx/htpasswd/' . ( $global ? 'default' : $this->db->site_url );
+		$format = EE\Utils\get_flag_value( $assoc_args, 'format' );
+		if ( $this->fs->exists( $file ) ) {
+			$user_lines = explode( PHP_EOL, trim( file_get_contents( $file ) ) );
+			foreach ( $user_lines as $line ) {
+				$users[]['users'] = strstr( $line, ':', true );
+			}
+
+			if ( 'text' === $format ) {
+				foreach ( $users as $user ) {
+					EE::log( $user['users'] );
+				}
+			} else {
+				$formatter = new EE\Formatter( $assoc_args, [ 'users' ] );
+				$formatter->display_items( $users );
+			}
+		} else {
+			EE::error( sprintf( 'http auth not enabled on %s', $this->db->site_url ) );
+		}
+	}
+
+	/**
+	 * create, append, remove, list ip whitelisting for a site or globally.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [<create>]
+	 * : Create ip whitelisting for a site or globally.
+	 *
+	 * [<append>]
+	 * : Append ips in whitelisting of a site or globally.
+	 *
+	 * [<list>]
+	 * : List whitelisted ip's of a site or of global scope.
+	 *
+	 * [<remove>]
+	 * : Remove whitelisted ip's of a site or of global scope.
+	 *
+	 * [<site-name>]
+	 * : Name of website to be secured / `global` for global scope.
+	 *
+	 * [--ip=<ip>]
+	 * : Comma seperated ips.
+	 */
+	public function whitelist( $args, $assoc_args ) {
+
+		$commands = [ 'create', 'append', 'list', 'remove' ];
+		if ( ! ( isset( $args[0] ) && in_array( $args[0], $commands ) ) ) {
+			$help = PHP_EOL;
+			foreach ( $commands as $command ) {
+				$help .= "ee auth whitelist $command [<site-name>/global] [--ip=<ip>]" . PHP_EOL;
+			}
+			EE::error( 'Please use valid command syntax. You can use:' . $help );
+
+		}
+
+		$command = array_shift( $args );
+		$global  = $this->populate_info( $args, __FUNCTION__ . ' ' . $command );
+
+		$ip = EE\Utils\get_flag_value( $assoc_args, 'ip' );
+
+		$file         = EE_CONF_ROOT . '/nginx/vhost.d/';
+		$file         .= $global ? 'default_acl' : $this->db->site_url . '_acl';
+		$user_ips     = array_filter( explode( ',', $ip ), 'strlen' );
+		$existing_ips = $this->get_ips_from_file( $global );
+
+		call_user_func_array( [ $this, "whitelist_$command" ], [ $file, $user_ips, $existing_ips ] );
```

Comment:
Add check if the method exists before calling it.

`if ( method_exists( [ $this, "whitelist_$command" ] ) )`

Reply:
Have added the check here: https://github.com/EasyEngine/auth-command/pull/1/files#diff-3881cdc58d8addcbed1dbb8e46eef342R177

Reply:
ok. no prob then. It's in the same class and the $command have a value from hardcode array.  Please add a comment on #176 that if we add something then it will need a method.

Reply:
Added comment